src/cascadia/TerminalSettingsEditor/AddProfile.cpp

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AddProfile.h"
+#include "AddProfile.g.cpp"
+#include "AddProfilePageNavigationState.g.cpp"
+#include "EnumEntry.h"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml::Navigation;
+using namespace winrt::Microsoft::Terminal::Settings::Model;
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    AddProfile::AddProfile()
+    {
+        InitializeComponent();
+    }
+
+    void AddProfile::OnNavigatedTo(const NavigationEventArgs& e)
+    {
+        _State = e.Parameter().as<Editor::AddProfilePageNavigationState>();

```

Comment:
if you add a profile and then go BACK to this page, does it contain the new profile?

if you change a profile name and go back to this page, does it show the new name?

Reply:
1. Yes
2. Yes (as long as the textbox lost focus at some point before you switched back to the add profile page)